materialize-motherduck/driver.go

```diff
@@ -229,26 +234,155 @@ func newTransactor(
 type binding struct {
 	target    sql.Table
 	storeFile *stagedFile
+	loadFile  *stagedFile
+	mustMerge bool
 }
 
 func (t *transactor) UnmarshalState(state json.RawMessage) error                  { return nil }
 func (t *transactor) Acknowledge(ctx context.Context) (*pf.ConnectorState, error) { return nil, nil }
 
 func (d *transactor) Load(it *m.LoadIterator, loaded func(int, json.RawMessage) error) error {
+	var ctx = it.Context()
+
 	for it.Next() {
-		panic("connector must be set to delta updates")
+		b := d.bindings[it.Binding]
+		b.loadFile.start()
+
+		if converted, err := b.target.ConvertKey(it.Key); err != nil {
+			return fmt.Errorf("converting Load key: %w", err)
+		} else if err = b.loadFile.encodeRow(ctx, converted); err != nil {
+			return fmt.Errorf("encoding Load key to scratch file: %w", err)
+		}
+	}
+	if it.Err() != nil {
+		return it.Err()
+	}
+
+	var subqueries []string
+	for _, b := range d.bindings {
+		if !b.loadFile.started {
+			// Pass.
+		} else if delete, err := b.loadFile.flush(); err != nil {
+			return fmt.Errorf("load.stage(): %w", err)
+		} else {
+			defer delete(ctx)
```

Comment:
this could defer multiple calls to `delete(ctx)`; is that intended?

Reply:
It should be one `delete(ctx)` for every binding `b` where `b.loadFile.started` is `true`, so it will defer a call to delete the staged file(s) for every binding that had keys to load documents for.

Reply:
:man_facepalming: I see it now; `delete` is a closure returned by `flush()`.

Might be worth a comment simply because it tripped up my "avoid defer in for loops" pattern recognizer.